File: 0001-0500/0119.Pascal's Triangle II.cpp
/**
 *Result:
 * 34 / 34 test cases passed.
 * Status: Accepted
 * Runtime: 0 ms
 * Your runtime beats 37.43% of cpp submissions.
 */
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<int> res;
        for(int i=0;i<=rowIndex;i++){
            int old =  1;
            for(int j=1;j<i;j++){
                int tmp = res[j];
                res[j] = old + res[j];
                old = tmp;
            }
            res.push_back(1);
        }
        return res;
    }
};

File: 0001-0500/0119.Pascal's Triangle II.py
class Solution(object):
    def getRow(self, k):
        """
        :type rowIndex: int
        :rtype: List[int]
        """
        # C_k_i = k! / (i! * (k - i)!)
        f = [1] * (k + 1)
        for i in xrange(1, k + 1):
            f[i] = f[i - 1] * i
        res = [None] * (k + 1)
        for i in xrange(k + 1):
            res[i] = f[k] // f[i] // f[k - i]
        return res
